internal/platform/select_darwin.s

```diff
@@ -0,0 +1,8 @@
+// lifted from golang.org/x/sys unix
+#include "textflag.h"
+
+TEXT libc_select_trampoline<>(SB), NOSPLIT, $0-0
+	JMP libc_select(SB)
+
+GLOBL ·libc_select_trampoline_addr(SB), RODATA, $8
+DATA ·libc_select_trampoline_addr(SB)/8, $libc_select_trampoline<>(SB)
```

Comment:
What is the reason we need this and cannot use `syscall.Select` for GOOS=darwin? https://pkg.go.dev/syscall?GOOS=darwin#Select

Reply:
one reason is that `syscall.Select` on darwin insists on _not_ returning an `(int, error)` pair and only giving the error https://github.com/golang/go/blob/release-branch.go1.20/src/syscall/syscall_darwin.go#L188; the other reason is that it wraps the `syscall.Errno` into an `error` (although that's less of an issue because we have `UnwrapOSError`)

Reply:
Even if `syscall.Select` returns a the number of events that completed, I believe the `select(2)` API still requires you to scan the whole fd set looking for file descriptors that are ready, so knowing how many there are does not really change the implementation isn't it?

Reply:
can you summarize these notes as comments into the .s file? it is hard to see these after the PR is merged.

Reply:
@achille-roussel AFAIU, it is common to check the result to be `>0`; for instance, compare @ktock's example in #1317. 

@codefromthecrypt I'll update the godoc on `syscall_select`

Reply:
actually I think the godoc says enough

Reply:
I think the point I'm trying to make is it's unfortunate to take on the maintenance of platform dependent assembly code just to get an integer returned that only saves us from scanning the fd set when there is a timeout. 

An alternative could be to implement platform.Select on Darwin by calling syscall.Select and scanning the fd sets to generate the count of ready fds.

I'm also wondering why Go chose not to return a count of ready fds on Darwin, I would be surprised if it was an error, so I'd assume there's a reason which is probably related portability (maybe in some versions of OSX or iOS the return value was broken?).

Now that you know everything I know, I'll get behind the choice you make whether it's keeping the code as-is or changing 😊

Reply:
well, we already have a precedent on Darwin for `utimens` where we needed to do it for a similar reason. Anyway, we can always dump and update this code, after all the goal of this PR was to get the approach right, I would be ok if we reverted to using `syscall.Select()` at any point.

as for why Go did not return the result in the first place, I don't know if it was a mistake, but I do know that on x/sys/unix it returns `(int, error)`

https://github.com/golang/sys/blob/master/unix/syscall_darwin.go#L615